### PR TITLE
Trigger service

### PR DESCRIPTION
**Pull request by gaborh-da**

### CODE REVIEW

File: scripts/startTriggers.sh [hunk 1]
```diff
@@ -22,29 +22,52 @@ fi
...
-      --trigger-name "$trigger_name" \
-      --ledger-host "${SANDBOX_HOST}" \
-      --ledger-port "${SANDBOX_PORT}" \
-      --ledger-party "$party"
+start_trigger() {
+  local pkg_id="$1"
+  local trigger_name="$2"
+  local party="$3"
+  curl --user ${party}':secret' \
+     -X POST localhost:8088/v1/start \
```
Comment (richardkapolnai-da) [inline scripts/startTriggers.sh:32]:
```suggestion
     -X POST localhost:$DEFAULT_SERVICE_PORT/v1/start \
```

Comment (gaborh-da) [reply scripts/startTriggers.sh]: Thanks, done

File: src/test/java/com/digitalasset/refapps/bondissuance/trigger/HttpClient.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HttpClient {
+  private final Logger logger = LoggerFactory.getLogger(getClass());
+
+  public boolean ping(URL url) {
```
Comment (richardkapolnai-da) [inline src/test/java/com/digitalasset/refapps/bondissuance/trigger/HttpClient.java:20]:
```suggestion
  public boolean isAvailable(URL url) {
```

File: src/test/java/com/digitalasset/refapps/bondissuance/triggerservice/TriggerService.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    this.logFile = new File(String.format("integration-test-trigger-service.log"));
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  private void start() throws Throwable {
+    ProcessBuilder processBuilder = createProcess();
+    processBuilder.environment().put("JAVA_TOOL_OPTIONS", "-Xmx350m");
```
Comment (richardkapolnai-da) [inline src/test/java/com/digitalasset/refapps/bondissuance/triggerservice/TriggerService.java:44]: why? can we remove this? if not, pls add a comment here

File: src/test/java/com/digitalasset/refapps/bondissuance/trigger/HttpClient.java (hunk 2 above)
Comment (gaborh-da) [reply src/test/java/com/digitalasset/refapps/bondissuance/trigger/HttpClient.java]: done

File: src/test/java/com/digitalasset/refapps/bondissuance/triggerservice/TriggerService.java (hunk 3 above)
Comment (gaborh-da) [reply src/test/java/com/digitalasset/refapps/bondissuance/triggerservice/TriggerService.java]: done

**Review by richardkapolnai-da — DISMISSED** [review]

**Review by richardkapolnai-da — APPROVED** [review]

**Pull request merged**